app/src/main/java/com/eveningoutpost/dexdrip/cgm/carelinkfollow/CareLinkFollowService.java

```diff
@@ -0,0 +1,293 @@
+            }
+            scheduleWakeUp();
+        } finally {
+            JoH.releaseWakeLock(wl);
+        }
+        return START_STICKY;
+    }
+
+    /**
+     * MegaStatus for Connect Follower
+     */
+    public static List<StatusItem> megaStatus() {
+        final BgReading lastBg = BgReading.lastNoSenssor();
+
+        long hightlightGrace = Constants.SECOND_IN_MS * 30; // 30 seconds
+
+        // Status for BG receive delay (time from bg was recorded till received in xdrip)
+        String ageOfBgLastPoll = "n/a";
+        StatusItem.Highlight ageOfLastBgPollHighlight = StatusItem.Highlight.NORMAL;
+        if (bgReceiveDelay > 0) {
+            ageOfBgLastPoll = JoH.niceTimeScalar(bgReceiveDelay);
+            if (bgReceiveDelay > SAMPLE_PERIOD / 2) {
+                ageOfLastBgPollHighlight = StatusItem.Highlight.BAD;
+            }
+            if (bgReceiveDelay > SAMPLE_PERIOD * 2) {
+                ageOfLastBgPollHighlight = StatusItem.Highlight.CRITICAL;
+            }
+        }
+
+        // Status for time since latest BG
+        String ageLastBg = "n/a";
+        StatusItem.Highlight bgAgeHighlight = StatusItem.Highlight.NORMAL;
+        if (lastBg != null) {
+            long age = JoH.msSince(lastBg.timestamp);
+            ageLastBg = JoH.niceTimeScalar(age);
+            if (age > SAMPLE_PERIOD + hightlightGrace) {
+                bgAgeHighlight = StatusItem.Highlight.BAD;
+            }
+        }
+
+        List<StatusItem> megaStatus = new ArrayList<>();
+
+        megaStatus.add(new StatusItem("Latest BG", ageLastBg + (lastBg != null ? " ago" : ""), bgAgeHighlight));
+        megaStatus.add(new StatusItem("BG receive delay", ageOfBgLastPoll, ageOfLastBgPollHighlight));
+        megaStatus.add(new StatusItem("Data period:", JoH.niceTimeScalar(SAMPLE_PERIOD)));
+        megaStatus.add(new StatusItem("Grace period:", JoH.niceTimeScalar(getGraceMillis())));
+        megaStatus.add(new StatusItem("Missed poll interval:", JoH.niceTimeScalar(getMissedIntervalMillis())));
+        megaStatus.add(new StatusItem());
+        megaStatus.add(new StatusItem("Last poll", lastPoll > 0 ? JoH.niceTimeScalar(JoH.msSince(lastPoll)) + " ago" : "n/a"));
+        megaStatus.add(new StatusItem("Last wakeup", last_wakeup > 0 ? JoH.niceTimeScalar(JoH.msSince(last_wakeup)) + " ago" : "n/a"));
+        megaStatus.add(new StatusItem("Next poll in", JoH.niceTimeScalar(wakeup_time - JoH.tsl())));
+        if (lastBg != null) {
+            megaStatus.add(new StatusItem("Last BG time", JoH.dateTimeText(lastBg.timestamp)));
+        }
+        megaStatus.add(new StatusItem("Last poll time", lastPoll > 0 ?  JoH.dateTimeText(lastPoll) : "n/a"));
+        megaStatus.add(new StatusItem("Next poll time", JoH.dateTimeText(wakeup_time)));
+        megaStatus.add(new StatusItem());
+        megaStatus.add(new StatusItem("Buggy Samsung", JoH.buggy_samsung ? "Yes" : "No"));
```

Comment:
Do you use the the Samsung workaround? If not, do not show it. It would confuse the users.

Reply:
I have added the buggy Samsung check, because it is used by all other similar follower service: `ShareFollowService`, `WebFollowService`, `NightscoutFollowService`

Reply:
Is it "used" by the other follower services or just shown? If it is not used by your service, please remove it.

Reply:
The `evaluate` method of `BuggySamsung` class is called at the beginning of `onStartCommand`, however I am not sure whether is has any effect or what was the purpose of it in the original similar services (`ShareFollowService` and `NightscoutFollowService`). As I wrote earlier I have used the logical structure of the similar follower services as templates and I didn't try to understand every little detail exactly. I can see, that even the new `WebFollowService` also contains this call and shows this information, so I believe there should be a reason for it.